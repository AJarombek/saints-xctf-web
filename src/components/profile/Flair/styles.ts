/**
 * JSS styles for the Flair component.
 * @author Andrew Jarombek
 * @since 9/8/2020
 */

import Colors from "../../../styles/colors";
import {FontMixins} from "../../../styles/mixins";

export default {
    flair: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        margin: '15px 0',

        '& > p': {
            textAlign: 'center',
            margin: '2px 0',
            color: Colors.sxctfRed,
            ...FontMixins.robotoSlabBold(),
            fontSize: 13
        }
    }
};
