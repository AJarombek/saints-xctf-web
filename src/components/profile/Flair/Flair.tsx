/**
 * Component on the profile page which shows the flair given to the user.
 * @author Andrew Jarombek
 * @since 9/8/2020
 */

import React, { memo } from 'react';
import { createUseStyles } from 'react-jss';
import styles from './styles';
import { FlairMeta } from '../../../redux/types';
import classNames from 'classnames';

interface Props {
  flair?: FlairMeta;
}

const useStyles = createUseStyles(styles);

const Flair: React.FunctionComponent<Props> = ({ flair = {} }) => {
  const classes = useStyles();

  return (
    <div className={classNames(classes.flair, 'flairList')}>
      {flair.items?.slice(0, 2).map((item) => (
        <p key={item.flair_id} data-cypress="flair">
          {item.flair}
        </p>
      ))}
    </div>
  );
};

export default memo<Props>(Flair);
