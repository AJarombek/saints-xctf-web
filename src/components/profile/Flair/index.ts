import Flair from "./Flair";

export default Flair;
