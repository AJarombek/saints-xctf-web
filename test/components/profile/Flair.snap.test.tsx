/**
 * Snapshot test for the Flair component.
 * @author Andrew Jarombek
 * @since 3/6/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import Flair from '../../../src/components/profile/Flair/Flair';

describe('Flair Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Flair
          flair={{
            items: [
              {
                flair: 'Sample Flair',
                flair_id: 1,
                username: 'andy'
              }
            ]
          }}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
