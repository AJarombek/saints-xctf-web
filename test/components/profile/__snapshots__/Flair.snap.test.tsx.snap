// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flair Snapshot Tests renders correctly 1`] = `
<div
  className="flair-0-1-1 flairList"
>
  <p
    data-cypress="flair"
  >
    Sample Flair
  </p>
</div>
`;
